cpp/include/cuspatial/polygon_distance.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required point_b_y applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+
+#include <memory>
+
+namespace cuspatial {
+
+/**
+ * @brief calculates minimum segment-point distance between all shapes
```

Comment:
```suggestion
 * @brief calculates minimum segment-to-point distance between all shapes
```

Reply:
And should it be "polygons" rather than "shapes", since that is specifically what this function supports? And "Euclidean distance" instead of just "distance"?

Reply:
The function supports individual points and/or segments, as well as polygons. Euclidean distance sounds good to me. :+1: